source/PlayerEntity.cpp

```diff
@@ -147,6 +155,58 @@ void PlayerEntity::onSimulation(SimTime absoluteTime, SimTime deltaTime) {
 			respawn();
 		}
 	}
+
+	if (!m_gettingMovementInput) {
+
+		if (accelerationEnabled!= nullptr && *accelerationEnabled) {
+			m_acceleratedVelocity = lerp(m_acceleratedVelocity, 0.0f, *movementDeceleration * deltaTime);
+		}
+		else {
+			m_acceleratedVelocity = 0;
+		}
+
+		if (m_acceleratedVelocity > 0) {
+			m_linearVector = m_acceleratedVelocity * m_lastDirection;
+		}
+
+		if (headBobEnabled!= nullptr && *headBobEnabled) {
+			m_headBobCurrentHeight = lerp(m_headBobCurrentHeight, 0.0f, 0.1f);
+		}
+	}
+	else {
+
+		if (*accelerationEnabled) {
+			m_acceleratedVelocity = lerp(m_acceleratedVelocity, m_walkSpeed, *movementAcceleration * deltaTime);
+		}
+		else {
+			m_acceleratedVelocity = m_walkSpeed;
+		}
+
+		m_linearVector = m_linearVector.direction() * m_acceleratedVelocity;
+		m_lastDirection = m_linearVector.direction();
+
+		if (headBobEnabled != nullptr && *headBobEnabled) {
+			if (!m_headBobPolarity) {
+				m_headBobCurrentHeight = lerp(m_headBobCurrentHeight, -*headBobAmplitude, *headBobFrequency * m_acceleratedVelocity * deltaTime);
```

Comment:
Again, it is not clear to me how this `lerp()` is working. I assume the theory is that it would scale with number of "footsteps per meter" which in turn scales with speed (higher speed ==> fewer steps per meter) but I'd like to either see more comments here explaining this or something introduced in the documentation describing how it relates.

Ideally you'd be able to set the head bob in a fixed unit that relates to speed (say bobs/(m/s)) and have this code be more deterministic, but for head bob I'm okay with a more fluid approach. Still I'm not sure this does what we want, more displacement (faster speed/longer sdt) implies stronger interpolation of the bob here where my intuition would be the opposite... Maybe I am confused.

Reply:
The amplitude is fixed (It will not go higher than what we set). The frequency of the head-bob will change based on players' velocity. So the faster the player moves, the faster his "head/camera" will oscillate. I have updated it with more comments. Let me know if I need to add a few more comments, break it down even further, or change the whole system all-together.

Reply:
I want to point out that the statement "it will not go higher than what we set" isn't necessarily true here. Again `lerp(a,b,c)` works using `a + (b-a)*c` so if `c` is >1 it will produce a value larger than `b` (in this case the `headBobAmplitude`).

Since your `c` (interpolation) term is `headBobFrequency  * acceleratedVelocity * deltaTime` its hard to say this quantity will always be <1. I'd prefer to move to a model here where we can safely guarantee this behavior across all possible values of `headBobFrequency` and `acceleratedVelocity` if possible.